src/CoWSwapEthFlow.sol

```diff
@@ -35,16 +36,35 @@ contract CoWSwapEthFlow is
     /// onchain data.
     mapping(bytes32 => EthFlowOrder.OnchainData) public orders;
 
-    /// @param _cowSwapSettlement The address of the CoW Swap settlement contract.
-    /// @param _wrappedNativeToken The address of the default native token in the current chain (e.g., WETH on mainnet).
+    /// @param _cowSwapSettlement The CoW Swap settlement contract.
+    /// @param _wrappedNativeToken The default native token in the current chain (e.g., WETH on mainnet).
     constructor(
         ICoWSwapSettlement _cowSwapSettlement,
-        IERC20 _wrappedNativeToken
+        IWrappedNativeToken _wrappedNativeToken
     ) CoWSwapOnchainOrders(address(_cowSwapSettlement)) {
         cowSwapSettlement = _cowSwapSettlement;
         wrappedNativeToken = _wrappedNativeToken;
+
+        _wrappedNativeToken.approve(
+            cowSwapSettlement.vaultRelayer(),
+            type(uint256).max
+        );
+    }
+
+    /// @inheritdoc ICoWSwapEthFlow
+    function wrap(uint256 amount) external {
+        wrappedNativeToken.deposit{value: amount}();
```

Comment:
is this new syntax? the `deposit{value: amount}` outside the parenthesis?

Reply:
This is indeed [new syntax since Solidity 0.7](https://docs.soliditylang.org/en/latest/070-breaking-changes.html?highlight=value%20gas%20call%20#changes-to-the-syntax). It seems reasonable to me as it makes clear the modifiers apply to the functions but aren't arguments.